### ISSUE TITLE
Conversion Netgen -> Medit -> Netgen

### ISSUE THREAD

**Issue by alessio-cesarano** [nschloe/meshio#1199]
I am trying to make a double conversion from a Netgen mesh (format .vol) to Medit Inria (.mesh) and back. The goal is to use an external library, mmg (https://www.mmgtools.org/) to optimize our mesh that we create with NGSolve and using it NGSolve later one.

Your software seems to be maybe the only one capable of such a conversion. While I successfully made the conversion in both ways, on the conversion back the information about material and interfaces gets lost (everything goes to 1 - default). I wanted to ask you if this is an issue you know about, or if I am doing something in a wrong way!

The commands I used are:
```
meshio convert -i netgen refined.vol -o medit refined.mesh
mmg3d_O3 -optim refined.mesh -out refined_optimized.mesh
meshio convert -i medit refined_optimized.mesh -o netgen refined_optimized.vol
```
I also double-checked if the mmg library was creating the problem by just doing a dummy conversion back of the same mesh, but I lose the information anyway:
```
meshio convert -i medit refined.mesh -o netgen refined_not_optimized.vol
```
Anyway I want to thank you for developing this very useful tool meshio and thank you in advance if you take the time to have a look at this issue!

Many thanks,
Alessio 

[meshes.zip](https://github.com/nschloe/meshio/files/7337676/meshes.zip)

**Comment by alessio-cesarano** [comment]
Hi Nico,

thank you very much for your quick reply! The original mesh is refined.vol. If I execute meshio info I get:

 meshio info refined.vol
 
<meshio mesh object>
  Number of points: 20932
  Number of cells:
    triangle: 33392
    tetra: 83360
  Cell data: netgen:index
  Field data: air, airgap, iron_stator, Wplus, Vminus, Uplus, Wminus, Vplus, Uminus, air_rotor, iron_rotor, default, interface_rotor, stator_inner, bottom, top, rotor_outer, stator_outer, outer, air_int_outer, air_int_inner
  
meshio info refined.mesh

<meshio mesh object>
  Number of points: 20932
  Number of cells:
    triangle: 33392
    tetra: 83360
  Point data: medit:ref
  Cell data: medit:ref

meshio info refined_not_optimized.vol 

<meshio mesh object>
  Number of points: 20932
  Number of cells:
    triangle: 33392
    tetra: 83360
  Cell data: netgen:index

**Comment by alessio-cesarano** [comment]
Hi Nico,

thank you for pointing out where to look. The good news is that I found the bug and a fix that works in my case:

I had to replace the label "netgen:index" at lines 367-371 of the file you mentioned [above!](https://github.com/nschloe/meshio/blob/main/src/meshio/netgen/_netgen.py#L364):

```
    cells_index = (
        mesh.cell_data["netgen:index"]
        if "netgen:index" in mesh.cell_data
        else [None] * len(mesh.cells)
        )
```

with "medit:ref":

```
    cells_index = (
        mesh.cell_data["medit:ref"]
        if "medit:ref" in mesh.cell_data
        else [None] * len(mesh.cells)
        )
```
This works because I am starting (in the second conversion), from a Medit format and I looked how you write the field information in the Medit format in the [_medit file!] (https://github.com/nschloe/meshio/blob/main/src/meshio/medit/_medit.py). I am not sure how you handle the data in structures so that it would work starting from any format.

Again, thank you very much for your work and your kindness!

Best,
Alessio
